bcgov/nr-spar
# Feat/Parent tree step foundation

# Description
Fixes #166 

This PR provides the ground work implementation of the parent tree step of Registration Flow.
The parent tree step is NOT yet finished.

### Testing guide
- Create a seedlot that has genetic worth shown this this [table](https://www2.gov.bc.ca/gov/content/industry/forestry/managing-our-forest-resources/tree-seed/seed-planning-use/spar#genetic_worth), I suggest using `FDC - Coastal Douglas-fir`
- Go to complete registration
- Go to Step 4 Orchard
- In the Orchard id field, type in `405` then wait for it to load
- Go to Step 5 Parent tree
- Table should be loaded (if the oracle-api is still working)
- Click on the eye icon to see special columns that you can toggle with in different tabs
- Table cell that has placeholder of `Add value` should accept user input and persist through tab changes
 
## Note:
This is not finished work so there might be some problems with some functionalities. I will be back to work on it.
The code in parent tree's `index.tsx` is quite spaghettish 🍝, the implementation will be improved and cleaned in the future. 
The registration flow tests are failing because `ReferenceError: structuredClone is not defined` from Jest, I could not figure this out in time, assistance is needed or we skip these tests

## Type of change

# How Has This Been Tested?
Manual as always

## Checklist







---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-182-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-182-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-182-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)
